frontend/src/_stores/dataQueriesStore.js

```diff
@@ -144,6 +164,11 @@ export const useDataQueriesStore = create(
             .finally(() => useAppDataStore.getState().actions.setIsSaving(false));
         },
         renameQuery: (id, newName) => {
+          /** If query creation in progress, skips call and pushes the update to queue */
+          if (get().creatingQueryInProcessId && get().creatingQueryInProcessId === id) {
```

Comment:
I guess we can remove get().creatingQueryInProcessId check and check for only equality like below

if (get().creatingQueryInProcessId === id) {
.....
}